src/NHibernate/Type/ManyToOneType.cs

```diff
@@ -224,5 +204,31 @@ public override bool[] ToColumnNullness(object value, IMapping mapping)
 			}
 			return result;
 		}
+
+		private bool IsDirtyManyToOne(object old, object current, bool[] checkable, ISessionImplementor session)
+		{
+			if (IsSame(old, current))
+			{
+				return false;
+			}
+
+			if (old == null || current == null)
+			{
+				return true;
+			}
+
+			if (ForeignKeys.IsTransientFast(GetAssociatedEntityName(), current, session) == true)
+			{
+				return true;
+			}
+
+			object oldid = GetIdentifier(old, session);
+			object newid = GetIdentifier(current, session);
+			IType identifierType = GetIdentifierType(session);
+
+			return checkable == null
+				? identifierType.IsDirty(oldid, newid, session)
+				: identifierType.IsDirty(old, current, checkable, session);
```

Comment:
Shouldn't it be `identifierType.IsDirty(oldid, newid, checkable, session);`?

Reply:
Ouch.. Indeed it should..